pkg/prowgen/prowgen.go

```diff
@@ -245,6 +245,10 @@ type generatePresubmitOptions struct {
 	disableRehearsal     bool
 }
 
+func (opts *generatePresubmitOptions) shouldAlwaysRun() bool {
```

Comment:
Having this on its own function is good, but in the same time it gives us the ability to test the behavior. Can you add a unittest with some basic test cases?